pyscf_ipu/nanoDFT/nanoDFT.py

```diff
@@ -140,10 +140,13 @@ class IterationState:
         ERI (TwoOrbitalMatrix): Two-electron repulsion integrals in the LCAO basis set.
         grid_weights (Array): Weights associated with the grid_AO
         mask (OrbitalVector): Orbital occupation mask.
-        input_floats (Array):
```

Comment:
129: change `num_orbitals` to N
125; add "." to end of sentence "mask."

Consider "OrbitalMatrixNxN" and "OrbitalVectorNx1" and "TwoOrbitalMatrixNxN" 

Thing the type can give us: 
(1) explain physics interpretation 
(2) explain shape
Do we want to do one or both? @awf apparently this arose from your suggestion to @hatemhelal : Did you mean (1), (2) or (1)+(2)?

Reply:
at least today I have been thinking that the variable name should give some of the physical interpretation and leaving the type annotation to describe the datatype and shape.  Examples:

* `FloatNxN` -> instead of `OrbitalMatrix` or `OrbitalMatrixNxN`
* `Float3` to describe a vector of exactly 3 floats

Further examples are on PR #87 

https://github.com/graphcore-research/pyscf-ipu/blob/83-THO/pyscf_ipu/experimental/types.py

Reply:
Looking at the before/after here, I do prefer `FloatNxN` over `OrbitalMatrix`